File: 1500/1431. Kids With the Greatest Number of Candies.go

func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := 0
	result := make([]bool, len(candies))
	for _, v := range candies {
		if v >= max {
			max = v
		}
	}
	for k, v := range candies {
		if v+extraCandies >= max {
			result[k] = true
		} else {
			result[k] = false
		}
	}
	return result
}

File: 1500/swift/1431. Kids With the Greatest Number of Candies.swift
class Solution {
    func kidsWithCandies(_ candies: [Int], _ extraCandies: Int) -> [Bool] {
        var maxCandy = 0
        var reslut: [Bool] = []
        for item in candies {
            if item > maxCandy {
                maxCandy = item
            }
        }
        for item in candies {
            if item + extraCandies >= maxCandy{
                reslut.append(true)
            }else {
                reslut.append(false)
            }
        }
        return reslut
    }
}